lib/index.js

```diff
@@ -5,7 +5,83 @@ const spawn = require('child_process').spawn;
 const spawnSync = require('child_process').spawnSync;
 const JSON = require('./json-buffer');
 const isElectron = require('is-electron');
+const fs = require('fs');
+const os = require('os');
+const index = [0];
+const timeStamp = new Date().toISOString();
+const net = require('net');
+// to hold the keep-alive connection to worker thread.
+let client;
+// set to true after keep-alive connection is successfully created.
+let clientConnected = false;
+const READFILETIMELIMIT = 30000;
+const debugMode = false;
 
+/**
+ * Update:
+ *    1. Instead of constantly relying on spawnSync for handling every msg, create a keep-alive connection that can be
+ *       used to send msg to worker thread.
+ *    2. For sending response back from worker thread, worker thread confirms the response file doesn't exist, and then
+ *       create the file and writes the response. Upon finishing writing, change the file permission to read-only.
+ *    3. For loading the response from main thread, check the existance of the response file and confirm the permission
+ *       is read-only, then read the file and delete it.
+ *    Step 2 & 3 serve as file locking mechanism so that the 2 processes can communicate through it.
+ */
+
+const responseFolder = path.join(os.tmpdir(), 'sync_rpc_responses');
```

Comment:
Should we have a suffix to the response folder ? Notebook process or just disconnecting and connecting with VsCode we will still use same folder and the response.txt

Reply:
Actually it should be fine if we want all the logs in the same file.